Initialise DLite search paths from entry points (#990)
SINTEF/dlite: doc/user_guide/search_paths.md

Search paths
============
It is possible to extend DLite with new datamodels, code generation templates and plugins, by appending to corresponding search path.  The table below lists the different types of search paths that are available in DLite.

| Search path type        | Description                                           |
|-------------------------|-------------------------------------------------------|
| storages                | Storage URLs or directory paths to datamodels         |
| templates               | Directory paths to code generator templates           |

> | storage_plugins         | Directory paths to storage plugins written in C       |

Reviewer: ```suggestion
| storage_plugins         | Directory paths to storage plugins (drivers) written in C       |
```

> | Search path type        | Description                                           |
> |-------------------------|-------------------------------------------------------|
> | storages                | Storage URLs or directory paths to datamodels         |
> | templates               | Directory paths to code generator templates           |
> | storage_plugins         | Directory paths to storage plugins written in C       |
> | mapping-plugins         | Directory paths to mapping plugins written in C       |
> | python-storage-plugins  | Directory paths to storage plugins written in Python  |
> | python-mapping-plugins  | Directory paths to mapping plugins written in Python  |
> | python-protocol-plugins | Directory paths to protocol plugins written in Python |

Reviewer: ```suggestion
| Search path type        | Description                                           |
|-------------------------|-------------------------------------------------------|
| storages                | Storage URLs or directory paths to datamodels         |
| templates               | Directory paths to code generator templates           |
| storage_plugins         | Directory paths to storage plugins (drivers) written in C       |
| mapping-plugins         | Directory paths to mapping plugins written in C       |
| python-storage-plugins  | Directory paths to storage plugins (drivers) written in Python  |
| python-mapping-plugins  | Directory paths to mapping plugins written in Python  |
| python-protocol-plugins | Directory paths to protocol plugins written in Python |
```

Reviewer: Sorry about the alignement at the right. It is better to do this in a proper editor. Here I can jsut guess,

| Search path type        | Environment variable name         | Python variable name              |
|-------------------------|-----------------------------------|-----------------------------------|
| storages                | DLITE_STORAGES                    | dlite.storage_path                |
| templates               | DLITE_TEMPLATE_DIRS               | dlite.template_path               |
| storage_plugins         | DLITE_STORAGE_PLUGIN_DIRS         | dlite.storage_plugin_path         |
| mapping-plugins         | DLITE_MAPPING_PLUGIN_DIRS         | dlite.mapping-plugin_path         |
| python-storage-plugins  | DLITE_PYTHON_STORAGE_PLUGIN_DIRS  | dlite.python-storage-plugin_path  |
| python-mapping-plugins  | DLITE_PYTHON_MAPPING_PLUGIN_DIRS  | dlite.python-mapping-plugin_path  |
| python-protocol-plugins | DLITE_PYTHON_PROTOCOL_PLUGIN_DIRS | dlite.python-protocol-plugin_path |

Setting environment variables
-----------------------------
This is typically done where you set up your environment, like in a virtualenv activate script or the users `~/.bashrc` file.

:::{note}
All the path variables, except for `DLITE_STORAGES`, uses (`:`) colon as path separator.

> However, since colon may appears in URLs, `DLITE_STORAGES` uses instead the pipe symbol (`|`) as path separator.

Reviewer: Why not have the same separator in all? Also, one could envison plugins being available on  the internet somewhere (for the future)?

:::

See [environment variables] for more details.

Appending to DLite path variables
---------------------------------
The `dlite` Python module defines the path variables listed in the table above.
A Python script or application can configure new datamodels and plugins by appending to these variables.

:::{example}
Adding the sub-directory `datamodels` to the search path for datamodels, can be done with:

```python
>>> from pathlib import Path
>>> import dlite
>>> dlite.storage_path.append(Path("my_package_root") / "datamodels")
```
:::

Using entry points
------------------
How to use entry points is easiest described with an example.

> Lets assume you have a package with the following directory layout:

Reviewer: ```suggestion
Let us assume you have a package with the following directory layout:
```
